pkg/filesystem/path/windows_parser.go

```diff
@@ -0,0 +1,152 @@
+package path
+
+import (
+	"strings"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type windowsParser struct {
+	path string
+}
+
+type windowsRelativeParser struct {
+	path string
+}
+
+// NewWindowsParser creates a Parser for Windows paths that can be used in Resolve.
+func NewWindowsParser(path string) (Parser, error) {
+	// Windows paths are generally passed to system calls that
+	// accept C strings. There is no way these can accept null
+	// bytes.
+	if strings.ContainsRune(path, '\x00') {
+		return nil, status.Error(codes.InvalidArgument, "Path contains a null byte")
+	}
+	return &windowsParser{path}, nil
+}
+
+// MustNewWindowsParser is identical to NewWindowsParser, except that it panics
+// upon failure.
+func MustNewWindowsParser(path string) Parser {
+	parser, err := NewWindowsParser(path)
+	if err != nil {
+		panic(err)
+	}
+	return parser
+}
+
+func (p windowsParser) ParseScope(scopeWalker ScopeWalker) (next ComponentWalker, remainder RelativeParser, err error) {
+	path := strings.Replace(p.path, "/", "\\", -1)
+
+	if len(path) >= 3 {
```

Comment:
Not too happy with this nested if. 
Maybe we should check for the empty string and let that exit early instead.
Then the is-absolute-branch can access the first character as well.

Reply:
Resolved in another thread.